VideoLocker/src/main/java/org/edx/mobile/view/CourseDetailTabActivity.java

```diff
@@ -51,12 +54,33 @@ protected void onCreate(Bundle savedInstanceState) {
 
         try{
             EnrolledCoursesResponse courseData = (EnrolledCoursesResponse) bundle
-                    .getSerializable(BaseFragmentActivity.EXTRA_ENROLLMENT);
-            activityTitle = courseData.getCourse().getName();
-            try{
-                segIO.screenViewsTracking(courseData.getCourse().getName());
-            }catch(Exception e){
-                logger.error(e);
+                    .getSerializable(Router.EXTRA_ENROLLMENT);
+
+            if ( courseData == null ){
+                //this is from notification
+                String courseId = bundle.getString(Router.EXTRA_COURSE_ID);
+                if (!TextUtils.isEmpty(courseId)){
+                    Api api = new Api(this);
+                    courseData = api.getCourseById(courseId);
+                    if (courseData != null && courseData.getCourse() != null ) {
+                        bundle.putSerializable(Router.EXTRA_ENROLLMENT, courseData);
+                        selectedTab = courseInfoTabIndex;
+                    }
+                }
+
+            }
+            if ( courseData != null ) {
```

Comment:
@aleffert  line 59  check if courseData from bundle is null  
                 line 72  check again because courseData maybe changed

I will add some comments